cmake/linker/ld/gcc/linker_flags.cmake

```diff
@@ -8,3 +8,7 @@ if (NOT CONFIG_COVERAGE_GCOV)
 endif()
 
 check_set_linker_property(TARGET linker APPEND PROPERTY gprof -pg)
+
+# GCC 11 by default emits DWARF version 5 which cannot be parsed by
+# pyelftools. Can be removed once pyelftools supports v5.
+add_link_options(-gdwarf-4)
```

Comment:
a late arrival comment, as I see this has already been merged.

This file is not intended to add symbols to any targets, it is intended to setup linker properties, and then the build system at another location decides if / when and where this property should be used.

So it should have been something like:
```
set_property(TARGET linker PROPERTY debug -gdwarf-4)
```

and then let other part of build system decide when to use it.

But as this is temporary solution, then no reason to make an update to this.

On the interesting part, Pyelftool 0.28 includes initial support for dwarf-5:
> + Version 0.28 (2022.02.03)
> ....
>     - Initial DWARF v5 support (#363 with several follow-ups)
https://github.com/eliben/pyelftools/pull/363

Reply:
> On the interesting part, Pyelftool 0.28 includes initial support for dwarf-5:

https://github.com/zephyrproject-rtos/zephyr/pull/46609. It turns out something is still broken even with pyelftools 0.29.

Reply:
> #46609. It turns out something is still broken even with pyelftools 0.29.

yes, from initial support in 0.28 and more fixes in 0.29 I do assume there is still some way to go.
Just wanted to raise awareness, but seems you already tried it out :+1: .
Guess I was still trying to get through my backlog at that time, cause I can't remember #46608 :smile: